tools/python/boutiques/publisher.py

```diff
@@ -215,9 +212,105 @@ def publish(self):
             if ret.upper() != "Y":
                 return
         self.zenodo_test_api()
+
+        # perform a search to check if descriptor is an updated version
+        # of an existing one
+        from boutiques.searcher import Searcher
+        searcher = Searcher(self.descriptor.get("name"), self.verbose,
+                            self.sandbox)
+        r = searcher.zenodo_search()
+
+        id_to_update = 0
+        for hit in r.json()["hits"]["hits"]:
+            title = hit["metadata"]["title"]
+            if title == self.descriptor.get("name"):
+                id_to_update = hit["id"]
+                break
+
+        if id_to_update:
+            if(not self.no_int):
+                prompt = ("Found an existing record with the same name, "
+                          "would you like to update it? "
+                          "(Y:update existing/n:publish new version)")
```

Comment:
I would say `"Publish new entry with name {}".format(name)` to make it clearer. Also, add a space after the closing bracket.